sw/device/lib/crypto/impl/status_unittest.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/impl/status.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+// NOTE: This test does not verify hardening measures; it only checks that the
+// "normal" contract of the functions is upheld.
+
+namespace status_unittest {
+namespace {
+
+TEST(Status, OkIsHardenedOk) {
+  EXPECT_EQ(hardened_status_ok(OTCRYPTO_OK), kHardenedBoolTrue);
+}
+
+TEST(Status, ErrorMacrosNotOk) {
+  // Error macros should evaluate to non-OK statuses.
+  EXPECT_EQ(status_ok(OTCRYPTO_BAD_ARGS), false);
+  EXPECT_EQ(status_ok(OTCRYPTO_RECOV_ERR), false);
+  EXPECT_EQ(status_ok(OTCRYPTO_FATAL_ERR), false);
+  EXPECT_EQ(status_ok(OTCRYPTO_ASYNC_INCOMPLETE), false);
+}
+
+TEST(Status, InterpretErrorMacros) {
+  // Error macros should translate to the crypto status implied by their name.
+  EXPECT_EQ(status_interpret(OTCRYPTO_OK), kCryptoStatusOK);
+  EXPECT_EQ(status_interpret(OTCRYPTO_BAD_ARGS), kCryptoStatusBadArgs);
+  EXPECT_EQ(status_interpret(OTCRYPTO_RECOV_ERR), kCryptoStatusInternalError);
+  EXPECT_EQ(status_interpret(OTCRYPTO_FATAL_ERR), kCryptoStatusFatalError);
+  EXPECT_EQ(status_interpret(OTCRYPTO_ASYNC_INCOMPLETE),
+            kCryptoStatusAsyncIncomplete);
+}
+
+static __attribute__((noinline)) crypto_status_t try_interpret(
```

Comment:
Nit: We're in an anonymous namespace, so you can drop `static`. You could probably even make this `constexpr`!